File: backend/app/schemas/validacion.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import VeredictoValidacionEnum


class ValidacionCrear(BaseModel):
    veredicto: VeredictoValidacionEnum
    pendientes: list[str] = []
    observaciones: str | None = None
    esterilizado: bool | None = None
    numero_microchip: str | None = None


class ValidacionSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    animal_id: int
    fecha: datetime
    veterinario: str
    veredicto: VeredictoValidacionEnum
    pendientes: list[str]
    observaciones: str | None

File: frontend/src/core/modelos/validacion.ts
import { VeredictoValidacion } from './enums';

export interface Validacion {
  id: number;
  animal_id: number;
  fecha: string;
  veterinario: string;
  veredicto: VeredictoValidacion;
  pendientes: string[];
  observaciones: string | null;
}

export interface ValidacionCrear {
  veredicto: VeredictoValidacion;
  pendientes: string[];
  observaciones: string | null;
  esterilizado: boolean | null;
  numero_microchip: string | null;
}
